pkg/network/opencontrail/cni/ipc/ipc.go

```diff
@@ -0,0 +1,326 @@
+	if err != nil {
+		log.Println("Error deleteing VM config file ", fname, " Error : ", err)
+		return err
+	}
+
+	return nil
+}
+
+func (conn *Connection) delVmToRpc(rpc *contrailDelRpc) error {
+	msg, err := json.MarshalIndent(rpc, "", "\t")
+	if err != nil {
+		return err
+	}
+
+	resp, err := conn.doOp("DELETE", msg)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		log.Println("Error in DEL VM message to VRouter. Code : ",
+			resp.StatusCode)
+		return fmt.Errorf("Error in DEL VM message to VRouter. Code : ",
+			resp.StatusCode)
+	}
+	return nil
+}
+
+func (conn *Connection) DelVm(podName string) error {
+	// Make RPC structure
+	rpc := makeDelRpc(podName)
+
+	// Remove the configuraion file stored for persistency
+	if err := conn.delVmToFile(rpc); err != nil {
+		return err
+	}
+
+	// Make the RPC call
+	if err := conn.delVmToRpc(rpc); err != nil {
+		return nil
+	}
+
+	return nil
+}
+
+/****************************************************************************
+ * POLL message handling
+ ****************************************************************************/
+type Result struct {
+	Vm   string `json:"vm"`
+	Ip   string `json:"ip-address"`
+	Plen int    `json:"plen"`
+	Gw   string `json:"gateway"`
+	Dns  string `json:"dns-server"`
+	Mac  string `json:"mac-address"`
+}
+
+type contrailVmRpcReq struct {
+	vm string `json:"vm"`
+}
+
+func initPollVmReq(instanceId string) *contrailVmRpcReq {
+	return &contrailVmRpcReq{vm: instanceId}
+}
+
+func (conn *Connection) pollVmOnce(instanceId string) (*Result, error) {
+	rpc := initPollVmReq(instanceId)
+	msg, err := json.MarshalIndent(rpc, "", "\t")
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := conn.doOp("GET", msg)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("Error polling VM. Status ", resp.StatusCode)
+	}
+
+	var result Result
+	var body []byte
+
+	body, err = ioutil.ReadAll(resp.Body)
+	log.Println("Read data ", string(body), err)
+	if err != nil {
+		return nil, err
+	}
+
+	err = json.Unmarshal(body, &result)
+	if err != nil {
+		return nil, err
+	}
+
+	return &result, nil
+}
+
+func (conn *Connection) GetVmiInfo(instanceId, ifName string,
+	delay string, count int) (*Result, error) {
+	for i := 0; i < count; i++ {
+		log.Println("Polling ", i)
+		result, err := conn.pollVmOnce(instanceId)
+		if err == nil {
+			return result, nil
+		}
+
+		log.Println("Error in Get ", err)
+		var d time.Duration
+		d, err = time.ParseDuration(delay)
+		if err != nil {
+			d, err = time.ParseDuration(delay)
+		}
+
+		time.Sleep(d)
+	}
+
+	return nil, fmt.Errorf("Failed in PollVM for instance %s", instanceId)
+}
+
+/****************************************************************************
+ * Connection handlers
+ ****************************************************************************/
+func (conn *Connection) Close() error {
```

Comment:
Close will be called from AddVm and DelVm APIs. This is changed in next commit. Right now, there is not much functionality there.